contrib/mpi-proxy-split/lower-half/libproxy.c

```diff
@@ -68,12 +68,22 @@ getDataFromMaps(const Area *text, Area *data, Area *heap)
       break;
     }
   }
-  // NOTE: Assume that data and heap are contiguous.
+  // NOTE: Assume that data, bss(if exists), and heap are contiguous.
+  /*
+   * The BSS section can be present on some platform and abscent on some.
+   * For e.g., the BSS section is abscent on Cori but present on SLES VM.
+   *
+   * The following code stops after heap region and saves BSS section address
+   * if it exists.
+   */
   void *heap_sbrk = sbrk(0);
   while (readMapsLine(mapsfd, &area)) {
     if (strstr(area.name, "[heap]") && area.endAddr >= (VA)heap_sbrk) {
       *heap = area;
       break;
+    } else {
+	// This should be the BSS section.
+        *bss = area;
```

Comment:
Don't need to break here?

Reply:
Is there a way to validate this assumption in the code that the segment following the data segment is the BSS? I mean instead of a blanket "else" case, can we add some additional check? This will save us a lot of debugging pain later if this assumption is violated on a different platform.

Reply:
Responding to @rohgarg :
> Don't need to break here? 
The `break` is required.  The logic is to set the bss section to the section just before the 'heap' section is found.  Please keep the 'break'.

Reply:
@rohgarg asks if we can add a check that we've found the bss section, to save possible painful debugging in the future if we fail to identify the bss section.  I agree.  Here is a possible suggestion in that direction.
    First, let's add a comment that we assume that prior to the heap, there are at most three segments (text, data, and bss)., or maybe also a read-only data segment   I suggest a stronger check as follows.

In the `txt` variable of type `Area`, after setting `txt.start = ...`, we can do:
1. Change `getTextSegmentRange()` to also save as an output parameter the (up to) 6 characters of `cmd` (aka `comm` in `man 5 proc`) in that `fscanf`.  (saved from /proc/*/stat; see `man 5 proc`)
2. After `txt.endAddr = (VA)end;`, add:
`strcpy(txt.name, cmd)` (but remove the open and close parentheses)
3. In the `if`, add `else if (strstr(area.name, txt.name)` and do `executableCounter++`.  (and initialize `executableCounter = 0`)
4. In the `else`, do `executableCounter = 0`
5. In the `if` condition that verifies that `heap` was seen, assert that `executableCounter >= 2` and `executableCounter <= 4`.